test/functional-e2e/user-management/user-profile-page-object.ts

```diff
@@ -25,25 +26,31 @@ const countryDropdown = 'div.MuiOutlinedInput-adornedEnd';
 const countryDropdownMenuSearch = 'div.MuiOutlinedInput-adornedEnd input';
 const countryDropdownMenuFirstOption = '.MuiAutocomplete-popper';
 const saveButtonUpdateProfilePage = 'button[type="submit"]';
-const closeButtonUpdateProfilePage = `.MuiGrid-container button[type="button"].ml-3 span`;
+const closeButtonUpdateProfilePage = `.MuiGrid-justify-content-xs-flex-end .MuiGrid-item button[type="button"] span`;
 const successMessage = '.MuiAlert-message';
-//const closeSuccessMessage = '.MuiAlert-message .close span:nth-child(1)';
+const closeSuccessMessage = 'button[aria-label="Close"] span svg';
 const addReferenceButton = '[title="Add a reference"] button';
 const referenceName = 'input[name="references.0.name"]';
 const referenceValue = 'input[name="references.0.uri"]';
-const removeReferenceButton = '.align-items-end button';
+const removeReferenceButton = 'button[title="Remove the reference"]';
 const userProilePageEntities = '.ct-card-body div div span';
 const spinner = '.spinner-grow';
 const userProfilePendingApplications =
-  '.d-flex .mt-2.ct-card:nth-child(3)  .ct-card-body div:nth-child(2 ) span span';
-const deleteApplicationButton = '.align-items-center button svg';
+  'div:nth-child(3).MuiBox-root  .ct-card-body div:nth-child(2 ) span span';
+const deleteApplicationButton =
+  'div:nth-child(3).MuiBox-root  .ct-card-body button';
+const userProfilePopup = 'div.MuiPopover-paper .MuiBox-root';
 
 export default class UserProfilePage {
   page: puppeteer.Page | undefined;
   value: string | undefined;
 
   async verifyUserProfileTitle(page: puppeteer.Page, username: string) {
-    await page.waitForSelector(userProfilePageName);
+    await waitForLoadingIndicatorToHide(page);
+    await page.waitForSelector(userProfilePageName, {
```

Comment:
as you have `waitForLoadingIndicatorToHide` util function, you can have `waitForSelectorToBe[Visible/Hidden]` or something similar